gdnative-core/src/object.rs

```diff
@@ -877,6 +877,34 @@ impl<'a, T: GodotObject, Access: ThreadAccess> TRef<'a, T, Access> {
     }
 }
 
+impl<'a, T, Access> TRef<'a, T, Access>
+where
+    T: GodotObject<RefKind = ManuallyManaged>,
+    Access: NonUniqueThreadAccess,
+{
+    /// Returns `true` if the pointer currently points to a valid object of the correct type.
+    /// **This does NOT guarantee that it's safe to use this pointer.**
+    ///
+    /// # Safety
+    ///
+    /// This thread must have exclusive access to the object during the call.
+    #[inline]
+    #[allow(clippy::trivially_copy_pass_by_ref)]
+    pub unsafe fn is_instance_sane(&self) -> bool {
```

Comment:
This method is not necessary on `TRef` because it is always assumed to be valid.